### ISSUE TITLE
Several broken links in backend/internal documentation

### ISSUE THREAD

**Issue by isislovecruft** [dalek-cryptography/curve25519-dalek#341]
#283 fixes a few of them but I think there's more since the repo restructuring last year. I don't have access to the buckets hosting the http://doc-internal.dalek.rs pages, so I'm not able to update them.

**Comment by pinkforest** [comment]
## Parallel doc
```
README.md:[parallel_doc]: https://doc-internal.dalek.rs/curve25519_dalek/backend/vector/avx2/index.html
src/lib.rs://! [parallel_doc]: https://doc-internal.dalek.rs/curve25519_dalek/backend/vector/avx2/index.html
```
## Questions / Findings
1) This is hard-coded to doc-internal.dalek.rs - not relative link
2) dalek.rs is just doing `make doc` and `make doc-internal` using nightly toolchain I'm assuming ?
3) Should this continue to rely on dalek.rs or should it moved to docs.rs generated as other crates ?

## make doc issue

\[target/doc/curve25519_dalek/\]backend/vector documentation doesn't get generated -

If continuing to use this method to generate docs is preferred - 

Problem is some documentation is behind some unstable feature flags

Which would need to be in RUSTDOCFLAGS which does not honor RUSTFLAGS which I didn't see set either

I can get the backend/vector to generate:

RUSTDOCFLAGS="-C target_feature=+avx2" cargo rustdoc --features "nightly std simd_backend" -- --html-in-header docs/assets/rustdoc-include-katex-header.html --document-private-items

Also I noticed that there are `#[cfg(rustdoc)]  in places but this isn't used anymore

This probably should move to `doc_cfg` pattern

```rust
#![feature(doc_cfg)]
#[doc(cfg(feature = "xx"))]
```

## docs.rs ?

Should the docs move to docs.rs ? For this would have to setup docsrs e.g.

```toml
[package.metadata.docs.rs]
# document all features
all-features = true
# defines the configuration attribute `docsrs`
rustdoc-args = ["--cfg", "docsrs"]

```

**Comment by pinkforest** [comment]
Fixed in #439

**Comment by pinkforest** [comment]
This was fixed via #439 @rozbb
